drafts/draft-20005.adoc

```diff
@@ -0,0 +1,225 @@
+
+Add the feature to the `server.xml`
+
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>ejbPersistentTimer-3.2</feature>
+    <feature>jdbc-4.2</feature>
+    ... other features
+  </featureManager>
+
+  <dataSource id="DefaultDataSource">
+    <jdbcDriver libraryRef="OraLib"/>
+    <properties.oracle URL="jdbc:oracle:thin:@//localhost:1521/EXAMPLEDB"/>
+    <containerAuthData user="dbuser" password="dbpwd"/>
+  </dataSource>
+  <library id="OraLib">
+    <file name="${shared.resource.dir}/jdbc/ojdbc8.jar" />
+  </library>
+
+  <!-- The following enables failover for persistent timers -->
+  <persistentExecutor id="defaultEJBPersistentTimerExecutor" missedTaskThreshold="5m"/>
+
+  ...
+</server>
+----
+
+[#CLM]
+== Load JAAS LoginModules from Resource Adapters
+
+Liberty supports JAAS LoginModules for configuring access to JCA managed resources. In general, these are packaged as a shared library and configured for Liberty. Sometimes the JCA ResourceAdapter packages a JAAS LoginModule as a part of it. In the past to use these JAAS LoginModules the classes from the ResourceAdapter had to be extracted and configured as a shared library, with this new feature the `jaasLoginModule` can now load the classes directly from the resource adapter simplifying the configuration. Before, it was necessary to package (or repackage) JAAS Custom Login Modules into a separate shared library in order to configure Liberty to use them. 
+
+Enable the AppSecurity and JCA features and configure a resource adapter. Use the `classProviderRef` attribute on the `jaasLoginModule` element to reference the `id` of the resource adapter.
+
+
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>appSecurity-2.0</feature>
+    <feature>jca-1.7</feature>
+    ... other features
+  </featureManager>
+
+  <resourceAdapter id="eciResourceAdapter" location="${shared.resource.dir}/cicseci.rar"/>
+
+  <!-- classProviderRef indicates that the login module class is found in the resource adapter -->
+  <jaasLoginModule id="identityProp" controlFlag="REQUIRED"
+      className="com.ibm.ctg.security.idprop.LoginModule"
+      classProviderRef="eciResourceAdapter">
+    <options propIdentity="Caller"/>
+  </jaasLoginModule>
+
+  <jaasLoginContextEntry id="CTGEntry" loginModuleRef="identityProp" name="CTGEntry"/>
+
+  <connectionFactory id="cf1" jndiName="eis/cf1" jaasLoginContextEntryRef="CTGEntry">
+    <properties.eciResourceAdapter ConnectionUrl="tcp://localhost" portNumber="2006" serverName="MYSERVER"/>
+  </connectionFactory>
+
+  ...
+</server>
+----
+
+The same approach can be used for JAAS Custom Login Modules that are packaged within an application. Set the `classProviderRef` to point to the `id` of the `application`, `webApplication`, or `enterpriseApplication` element that contains the login module class. When packaging JAAS Custom Login Modules within an application, include the login module within one of the following places:
```

Comment:
I think that JAAS Custom Login Modules can be "JASS custom login modules" in every instance.